### PR TITLE
[O2B-1315] Consume ECS kafka messages to get runs and environment information

### PR DESCRIPTION
**Pull request by martinboulais**
#### I have a JIRA ticket
- [X] branch and/or PR name(s) include(s) JIRA ID
- [X] issue has "Fix version" assigned
- [X] issue "Status" is set to "In review"
- [X] PR labels are selected

Notable changes for users:
- Consume ECS kafka messages to get runs and environment information

### CODE REVIEW

File: lib/server/kafka/AliEcsEventMessagesConsumer.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                    EventMessage.decode(message.value),
+                    { enums: String },
+                ));
+            },
+        });
+    }
+
+    /**
+     * Handle a received message
+     *
```
Comment (xsalonx) [inline lib/server/kafka/AliEcsEventMessagesConsumer.js:92]: I would say it `calls each listener with given message` or anything more specific that 'Handle message'

File: lib/server/kafka/AliEcsSynchronizer.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            if (state === 'CONFIGURED' && transition === 'START_ACTIVITY') {
+                runService
+                    .createOrUpdate(runNumber, environmentId, { timeO2Start: timestamp.toNumber() })
+                    .catch((error) => this._logger.errorMessage(`Failed to save run start for ${runNumber}: ${error.message}`));
+            }
+
+            if (state === 'RUNNING' && transition === 'STOP_ACTIVITY') {
+                runService
+                    .createOrUpdate(runNumber, environmentId, { timeO2End: timestamp.toNumber() })
+                    .catch((error) => this._logger.errorMessage(`Failed to save run start for ${runNumber}: ${error.message}`));
```
Comment (xsalonx) [inline lib/server/kafka/AliEcsSynchronizer.js:67]: `Failed to save run end ...` ?

File: lib/server/services/environment/EnvironmentService.js [hunk 3]
```diff
@@ -117,6 +117,52 @@ class EnvironmentService {
         return this.get(environmentId);
     }
 
+    /**
+     * Define environment state at a given time
+     *
+     * @param {string} environmentId the id of the environment for which a new state must be registered
+     * @param {EnvironmentState} state the state to save
+     * @param {object<string, string>} vars updated environment vars
+     * @return {Promise<void>} resolves once the state has been saved
```
Comment (xsalonx) [inline lib/server/services/environment/EnvironmentService.js:126]: `vars` ?

File: lib/server/services/environment/EnvironmentService.js [hunk 4]
```diff
@@ -117,6 +117,52 @@ class EnvironmentService {
...
+                    }
+                }
+                await this.update(
+                    environmentId,
+                    {
+                        updatedAt: state.timestamp,
+                        rawConfiguration: JSON.stringify(configuration),
+                    },
+                    state,
+                );
```
Comment (xsalonx) [inline lib/server/services/environment/EnvironmentService.js:161]: If for some reason the try-catch block fails, `configuration` remains `undefined`, what can happen only if `rawConfiguration` is not valid json, then you update invalid rawConfiguration with `undefined`. Is it intended?

File: lib/server/services/run/RunService.js [hunk 5]
```diff
@@ -301,6 +377,85 @@ class RunService {
             .findAll(queryBuilder))
             .map((row) => ({ l3Level: row.get('l3Level'), dipoleLevel: row.get('dipoleLevel') }));
     }
+
+    // eslint-disable-next-line valid-jsdoc
```
Comment (xsalonx) [inline lib/server/services/run/RunService.js:381]: Why disabled?

File: lib/server/services/run/RunService.js [hunk 6]
```diff
@@ -280,6 +311,51 @@ class RunService {
...
+            },
+            [runNumber],
+            tags,
+            [],
+            [],
+            [],
+            transaction,
+        );
+        if (error) {
+            // TODO log the failed log creation
```
Comment (xsalonx) [inline lib/server/services/run/RunService.js:355]: Not in this PR?

**Review by xsalonx — CHANGES_REQUESTED** [review]

File: lib/server/kafka/AliEcsEventMessagesConsumer.js (hunk 1 above)
Comment (martinboulais) [reply lib/server/kafka/AliEcsEventMessagesConsumer.js:92]: Agreed

File: lib/server/kafka/AliEcsSynchronizer.js (hunk 2 above)
Comment (martinboulais) [reply lib/server/kafka/AliEcsSynchronizer.js:67]: Indeed

File: lib/server/services/environment/EnvironmentService.js (hunk 3 above)
Comment (martinboulais) [reply lib/server/services/environment/EnvironmentService.js:126]: Named vars in ECS, but I agree with you, renamed

File: lib/server/services/run/RunService.js (hunk 5 above)
Comment (martinboulais) [reply lib/server/services/run/RunService.js:381]: For some reason it do not accept the lambda function description

File: lib/server/services/run/RunService.js (hunk 6 above)
Comment (martinboulais) [reply lib/server/services/run/RunService.js:355]: Agreed

File: lib/server/services/run/RunService.js (hunk 5 above)
Comment (xsalonx) [reply lib/server/services/run/RunService.js:381]: what about `function<Error, void>`

Comment (martinboulais) [reply lib/server/services/run/RunService.js:381]: For my IDE it do not work?

File: lib/server/services/environment/EnvironmentService.js (hunk 4 above)
Comment (martinboulais) [reply lib/server/services/environment/EnvironmentService.js]: `JSON.stringify(undefined)` returns undefined

**Pull request merged**